### ISSUE TITLE
Mac won't shutdown or reboot with eGPU connected

### ISSUE THREAD

**Issue by whoacowboy** [goalque/automate-eGPU#21]
I am having an issue with getting my eGPU working.

I have tried this with my newer MacPro and MacAir. I can get both machines to recognize the eCPU, but neither of them will shutdown or boot if the thunderbolt cable is connected. I was able to get an external monitor to work on my laptop if I did the following.

Boot my laptop with the thunderbolt cable disconnected.
At the login screen connect my thunderbolt cable.
Sign in.
Everything loads up, yeay.
Restart and the computer hangs.

I haven't gone through this process with the desktop, but I am hopeful.

Has anybody else seen this, or does anybody have any ideas?

**Comment by goalque** [comment]
On OS X? Please provide more detail.

- OS X build version
- A single or multi-slot enclosure
- MacBook Air model identifier
- eGPU

**Comment by whoacowboy** [comment]
Thanks for the quick response.

  Model Name:	Mac Pro
  Version 10.11.2 (15C50)
  Model Identifier:	MacPro6,1

  Model Name:	MacBook Air
  Version: 10.11.3 (15D21)
  Model Identifier:	MacBookAir6,2

  Vendor Name:	Sonnet Technologies, Inc.
  Device Name:	Echo Express III TB2
  3 Slots only one fits the card.
  
  NVIDIA Quadro K5000 for MAC 4GB GDDR5 Graphics card (PNY Part #: VCQK5000MAC-PB)

**Comment by goalque** [comment]
That’s an accurate list :)

The first thing you can try is "sudo ./automate-eGPU.sh -a"
and restart the Mac.

Quadro K5000 is a Kepler-based GPU and even though -skipdriver option may work, it’s highly recommended to use the latest web driver from Nvidia by running the script without arguments, especially on OS X El Capitan.

If the web driver is installed manually, uninstall everything: sudo ./automate-eGPU.sh -uninstall. Then restart the Mac, rerun the script with the eGPU plugged, turn on -a mode, and restart.

Connect your eGPU to a port 5 or 6 of the nMP:
https://support.apple.com/en-us/HT202801

**Comment by whoacowboy** [comment]
Sorry this is on hold, I installed the graphics card on an old mac pro to render something for a deadline. 

I will try again once that is done. 

I do have a side question. I am doing Nvidia Iray renders from my mac and I want to get the largest card that I can staying on a native mac os x (no bootcamp). Any ideas what nvidia cards are compatible, or even what types of cards I should look at?

**Comment by goalque** [comment]
No problem.

I don’t know anything about Nvidia Iray. I’m waiting for the Indigo 4 and its multi-GPU support. It uses OpenCL and AMD performs better in that area.

**Comment by whoacowboy** [comment]
OK, i got this to work and then it stopped woking.

Apparently the 

/System/Library/Extensions/NVDAStartup.kext
/System/Library/Extensions/AppleHDA.kext/Contents/PlugIns/AppleHDAController.kext

My updates are being overwritten on every 2nd or 3rd reboot. 

This happened with the automated option and manual.

When I manually change them back all is good again.

I can open a new issue if you want, because even though it is the same issue, it is a different problem.

**Comment by goalque** [comment]
The v0.9.8 doesn't touch AppleHDA.kext. You have modified kexts manually? That's the problem. Please reinstall OS X, and never modify kexts manually.

**Comment by whoacowboy** [comment]
Hey @goalque do I need a complete reinstall it can I just restore the original ktext?

**Comment by irozhkov** [comment]
restore with correct permissions on file

**Comment by goalque** [comment]
OS X reinstallation from recovery preserves your files and apps (backup them anyway). A time machine restore is possible as well. The last option is erase & reinstall.

Script's -uninstall option won’t return the original kexts if some of them were modified before the first execution of the script.

**Comment by whoacowboy** [comment]
I initially followed the instructions [here](http://forum.netkas.org/index.php?topic=11166.0) which had me editing the wrong file. I didn't reinstall the os, but here is what I did in case it helps anybody.

- unplugged thunderbolt to eGPU, TB2.
- started my machine.
- removed my manual updates.
- ran `sudo ./automate-eGPU.sh -uninstall`
- restarted machine (TB2 still unplugged).
- hot plugged TB2.
- ran `sudo ./automate-eGPU.sh`
- restarted computer while crossing fingers

![screen shot 2016-04-12 at 12 18 12 pm](https://cloud.githubusercontent.com/assets/8116953/14472579/aaf60b4a-00a8-11e6-9fa9-45a057dd3354.png)

Thanks @goalque @irozhkov for your help!

**Comment by whoacowboy** [comment]
OK, so this didn't work. The computer would hang on boot. 

I did follow @goalque 's direction and the box just started up. 

- unplugged thunderbolt to eGPU, TB2.
- i did a recovery install of OS X.
- rebooted
- ran `sudo ./automate-eGPU.sh -uninstall`
- restarted machine (TB2 still unplugged).
- hot plugged TB2.
- ran `sudo ./automate-eGPU.sh`
- restarted computer it froze
- rebooted with TB2 unplugged
- ran `sudo ./automate-eGPU.sh -a`
- restarted and it worked
- restarted one more time for good measure and it worked, it did take longer to boot up though

Thanks again.

**Comment by goalque** [comment]
Yep, the -a mode is often necessary with multi-slot enclosures.

**Comment by agosto-levitomes** [comment]
I have a similar issue on my 2013 mbp/ Echo Express III.
OSX V110.11.3
Nvidia 980 TI

Restarts cause the laptop to hang/100% fan/very hot.
Sometimes (often) on boot the same thing happens. I have to disconnect, turn off, reconnect, and turn on everything in a particular order.

I assumed this was pretty normal?
I get the same results with an AKiTiO enclosure.

**Comment by goalque** [comment]
The -a mode may cause heat issues, but you can turn it off after successful boot:
sudo ./automate-eGPU.sh -m

this removes the background daemon process which is only necessary for the next reboot.

**Comment by whoacowboy** [comment]
@agosto-levitomes I don't know if you tried other solutions before, I did, and that is what caused a lot of the issues. Doing the recovery install of OS X, and going through the process again fixed it. Back everything up, but all my apps, files and the majority of my settings were untouched.

**Comment by goalque** [comment]
@whoacowboy thanks for the clarification.

**Comment by whoacowboy** [comment]
@goalque 
No problem, what you are doing is really helpful and I want it to keep going!
